From da8f894d97ac2d84e8e08e17efc1eb78e20169a3 Mon Sep 17 00:00:00 2001
From: Nataliia Cherepanova <nataly.cher@partner.samsung.com>
Date: Wed, 16 Aug 2023 11:43:21 +0300
Subject: [PATCH 163/225] [Unify Software Stack] Create install_pnm_resource.sh
 script

Also MAINTAINERS and PNM_HOW_TO_BUILD.md updated

Refers to MCS23-1024

Signed-off-by: Nataliia Cherepanova <nataly.cher@partner.samsung.com>
---
 MAINTAINERS             |  2 ++
 PNM_HOW_TO_BUILD.md     | 11 +++++++++++
 install_pnm_resource.sh | 12 ++++++++++++
 3 files changed, 25 insertions(+)
 create mode 100644 install_pnm_resource.sh

diff --git a/MAINTAINERS b/MAINTAINERS
index 7851900b6..202cb6b54 100644
--- a/MAINTAINERS
+++ b/MAINTAINERS
@@ -5237,9 +5237,11 @@ S:	Maintained
 F:	PNM_HOW_TO_BUILD.md
 F:	config-pnm
 F:	drivers/pnm/
+F:	include/uapi/linux/pnm_resources.h
 F:	include/uapi/linux/imdb_resources.h
 F:	include/uapi/linux/libaxdimm.h
 F:	include/uapi/linux/sls_resources.h
+F:	install_pnm_resource.sh
 F:	install_sls_headers.sh
 F:	install_sls_resource.sh
 F:	install_dax_drivers.sh
diff --git a/PNM_HOW_TO_BUILD.md b/PNM_HOW_TO_BUILD.md
index c163c16e9..19d356be4 100644
--- a/PNM_HOW_TO_BUILD.md
+++ b/PNM_HOW_TO_BUILD.md
@@ -53,6 +53,17 @@ Built drivers can then be installed with `install_dax_drivers` script:
 $ ./install_dax_drivers.sh
 ```
 
+# Class resource driver only (required to install SLS/IMDB resource)
+
+```bash
+$ make M=drivers/pnm/ -j32
+```
+
+Built drivers can then be installed with `install_pnm_resource` script:
+```bash
+$ ./install_pnm_resource.sh
+```
+
 # SLS resource driver only
 
 ```bash
diff --git a/install_pnm_resource.sh b/install_pnm_resource.sh
new file mode 100644
index 000000000..ca3b37d92
--- /dev/null
+++ b/install_pnm_resource.sh
@@ -0,0 +1,12 @@
+#!/bin/sh
+# SPDX-License-Identifier: GPL-2.0
+# Copyright(c) 2023 Samsung LTD. All rights reserved.
+
+set -e
+
+sudo cp -v drivers/pnm/*.ko /lib/modules/`uname -r`/kernel/drivers/pnm/
+sudo cp -v include/uapi/linux/pnm_resources.h /usr/include/linux
+sudo cp -v include/uapi/linux/pnm_resources.h /usr/src/linux-headers-`uname -r`/include/uapi/linux
+
+lsmod | grep pnm_class_resource > /dev/null && sudo rmmod pnm_class_resource
+sudo modprobe -v pnm_class_resource "$@"
-- 
2.34.1

